### PR TITLE
[WIP] Mergin project panel

### LINKED ISSUE #175: Change MY PROJECTS and ALL PROJECTS
See the gdoc for more info.

Or something along those lines.  A couple of minor improvements:
 - [x] using different tabs for local, private, shared with me and public project
 - [x] adding search project toolbar
 - [x] adding user profile info (show mergin storage, etc)
- [x] update api with namespace according API (download, fetch, upload, etc..)

### PR DESCRIPTION
**Pull request by sklencar**
This PR relies on current Mergin phase release

closes #175

### CODE REVIEW

File: app/inputsearchmodel.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "inputsearchmodel.h"
+
+InputSearchModel::InputSearchModel( QObject *parent )
```
Comment (PeterPetrik) [inline app/inputsearchmodel.cpp]: `=default` instead of brackets

File: app/inputsearchmodel.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#ifndef INPUTSEARCHMODEL_H
```
Comment (PeterPetrik) [inline app/inputsearchmodel.h]: add copyright header

File: app/inputsearchmodel.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#include "inputsearchmodel.h"
```
Comment (PeterPetrik) [inline app/inputsearchmodel.cpp]: add copyright header

File: app/merginapi.cpp [hunk 4]
```diff
@@ -22,7 +22,7 @@ MerginApi::MerginApi( const QString &dataDir, QObject *parent )
   loadAuthData();
 }
 
-void MerginApi::listProjects( const QString &filterTag )
+void MerginApi::listProjects( const QString &searchExpression, const QString &user, const QString &flag, const QString &filterTag )
```
Comment (PeterPetrik) [inline app/merginapi.cpp]: line too long, split args to new lines

File: app/merginapi.h [hunk 5]
```diff
@@ -108,6 +116,13 @@ class MerginApi: public QObject
     QString apiRoot() const;
     void setApiRoot( const QString &apiRoot );
 
+    QString searchExpression() const;
+    void setSearchExpression( const QString &searchExpression );
+
+    int diskUsage() const;
```
Comment (PeterPetrik) [inline app/merginapi.h]: add docs .. this is in MB or bytes or ?

File: app/merginapi.h [hunk 6]
```diff
@@ -108,6 +116,13 @@ class MerginApi: public QObject
...
+
+    int storageLimit() const;
```
Comment (PeterPetrik) [inline app/merginapi.h]: add docs .. this is in MB or bytes or ?

File: app/merginapi.h [hunk 7]
```diff
@@ -54,9 +57,13 @@ class MerginApi: public QObject
      * when a response is received, parses project json, writes it to a cache text file and sets mMerginProjects.
      * Eventually emits listProjectsFinished on which ProjectPanel (qml component) updates content.
      * If listing has been successful, updates cached merginProjects list.
+     * @param searchExpression Search filter on projects name.
```
Comment (PeterPetrik) [inline app/merginapi.h]: don't we use `\param` variant for doxy commands?

File: app/projectsmodel.h [hunk 8]
```diff
@@ -26,7 +27,7 @@
  * and create list model from them. Available are full path to the file, name of the project
  * and short name of the project (clipped to N chars)
  */
-class ProjectModel : public QAbstractListModel
+class ProjectModel : public InputSearchModel
```
Comment (PeterPetrik) [inline app/projectsmodel.h]: if `InputSearchModel` is just container for one string value, why this value cannot be just added to ProjectModel? or does the inputSearchModel has any logic?

Comment (sklencar) [reply app/projectsmodel.h]: There are two models - ProjectsModel and MerginProjectsModel. I wanted to have search string in both. However, MerginProjectsModel can be simplified and work without searchString

File: app/inputsearchmodel.cpp (hunk 1 above)
Comment (sklencar) [reply app/inputsearchmodel.cpp]: Default constructor cannot have arguments. The InputSearchModel is inherited from QAbstractListModel where a constructor has param `QObject *parent`

**Comment by PeterPetrik** [comment]
can you create new PR with 1-2 commits with nice commit messages from lutra/input so we have APK?
 and close this one?

**Comment by sklencar** [comment]
Closing on request. See PR #292

**Pull request closed without merging**